var/spack/repos/builtin/packages/pfunit/package.py

```diff
@@ -61,10 +74,17 @@ def cmake_args(self):
             '-DMAX_RANK=%s' % spec.variants['max_array_rank'].value]
 
         if spec.satisfies('+mpi'):
-            args.extend(['-DMPI=YES', '-DMPI_USE_MPIEXEC=YES',
+            if spec.satisfies('@4.0.0:'):
+                args.append('-DSKIP_MPI=NO')
+            else:
+                args.append('-DMPI=YES')
+            args.extend(['-DMPI_USE_MPIEXEC=YES',
```

Comment:
Looks good.  My only suggestion would be to reverse the nesting of `spec.satisfies('+mpi')` and `spec.satisfies('@4.0.0:'`.  I think it will be easier to follow the logic for a particular version if the version check is the outer `if` statement.

Reply:
Sure, I can take care of that right now.  Thank you for the reviews!

Reply:
Actually, this will complicate the logic of

```
            args.extend(['-DMPI_USE_MPIEXEC=YES',
                         '-DCMAKE_Fortran_COMPILER=%s' % spec['mpi'].mpifc])
```
which happens for +mpi no matter the version.  Would you still like this change?